base/logger_file.go

```diff
@@ -384,3 +396,27 @@ func runLogDeletion(ctx context.Context, logDirectory string, logLevel string, s
 
 	return nil
 }
+
+// getDeletionDirAndRegexp will return the directory and a regexp matching log file and rotated patterns.
+func getDeletionDirAndRegexp(filename string) (string, *regexp.Regexp) {
+	dir, path := filepath.Split(filename)
+	lastDot := strings.LastIndex(path, ".")
+	if lastDot == -1 {
+		// foo
+		// foo-2019-01-01T00-00-00.000
+		// foo-2019-01-01T00-00-00.000.gz
+		return dir, regexp.MustCompile(fmt.Sprintf(`(%s|%s%s(?:\.gz))?$`, regexp.QuoteMeta(path), regexp.QuoteMeta(path), lumberjackRotationMidfix))
+	}
+	prefix := path[:lastDot]
+	suffix := path[lastDot:]
+	// foo.log
+	// foo-2019-01-01T00-00-00.000.log
+	// foo-2019-01-01T00-00-00.000.log.gz
+	//
+	// or
+	//
+	// foo.bar.log
+	// foo.bar-2019-01-01T00-00-00.000.log
+	// foo.bar-2019-01-01T00-00-00.000.log.gz
+	return dir, regexp.MustCompile(fmt.Sprintf(`(%s|%s%s%s(?:\.gz)?)$`, regexp.QuoteMeta(path), regexp.QuoteMeta(prefix), lumberjackRotationMidfix, regexp.QuoteMeta(suffix)))
```

Comment:
Took me a while to separate the Sprintf and regexp syntax and I think it would be slightly more readable without the mix, especially if we do `regexp.QuoteMeta` outside of the pattern building, and build up the rotated pattern before passing it into the larger pattern.

Doing this allows us to use the same pattern for both formats, by having `rotatedPattern` built for the LastIndex case

```suggestion
	filenamePattern := regexp.QuoteMeta(filename)
	rotatedPattern := regexp.QuoteMeta(prefix) + lumberjackRotationMidfix + regexp.QuoteMeta(suffix) + `(\.gz)?`
	return dir, regexp.MustCompile(`^(` + filenamePattern + ` | ` + rotatedPattern + `)$`)
```

Reply:
This looks great, thanks!!